### PR TITLE
Job backend

### PR DESCRIPTION
**Pull request by KitNaIn**
Simple Backend for Jobs added frontend dashboard

### CODE REVIEW

File: backend/src/test/java/de/neuefische/backend/controller/JobControllerTest.java [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+    private ObjectMapper objectMapper;
+
+    @Autowired
+    private JobService jobService;
+
+    @Autowired
+    private JobRepo jobRepo;
+
+    @Autowired
+    private GenerateUUIDService generateUUIDService;
```
Comment (Kreyno93) [inline backend/src/test/java/de/neuefische/backend/controller/JobControllerTest.java:44]: no need for that anymore.
use lombok.config
and the tests will be much slimmer

File: backend/src/test/java/de/neuefische/backend/model/JobTest.java [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+        assertEquals(newEndTime, job.getEndTime());
+
+        String newStatus = "New Status";
+        job.setStatus(newStatus);
+        assertEquals(newStatus, job.getStatus());
+
+        String newJobComment = "New Comment";
+        job.setJobComment(newJobComment);
+        assertEquals(newJobComment, job.getJobComment());
+    }
```
Comment (Kreyno93) [inline backend/src/test/java/de/neuefische/backend/model/JobTest.java:79]: can be removed after config

File: backend/src/test/java/de/neuefische/backend/model/JobTest.java [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+    @Test
+    void testNoArgsConstructorSetter() {
+        //GIVEN
+        Job job = new Job();
+        //WHEN
+        job.setIssuer("Issuer");
+        //THEN
+        assertEquals("Issuer", job.getIssuer());
+    }
```
Comment (Kreyno93) [inline backend/src/test/java/de/neuefische/backend/model/JobTest.java:95]: same

File: backend/src/test/java/de/neuefische/backend/service/GenerateUUIDServiceTest.java [hunk 4]
```diff
@@ -0,0 +1,22 @@
+package de.neuefische.backend.service;
+
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.test.context.junit.jupiter.SpringExtension;
+
+import static org.junit.jupiter.api.Assertions.*;
+@ExtendWith(SpringExtension.class)
+@SpringBootTest
```
Comment (Kreyno93) [inline backend/src/test/java/de/neuefische/backend/service/GenerateUUIDServiceTest.java:10]: no need in unit tests

File: backend/src/test/java/de/neuefische/backend/service/JobServiceTest.java [hunk 5]
```diff
@@ -0,0 +1,200 @@
...
+import org.mockito.Mockito;
+import org.springframework.boot.test.context.SpringBootTest;
+
+import java.time.LocalDate;
+import java.time.LocalTime;
+import java.util.ArrayList;
+import java.util.List;
+
+import static org.junit.jupiter.api.Assertions.*;
+@SpringBootTest
```
Comment (Kreyno93) [inline backend/src/test/java/de/neuefische/backend/service/JobServiceTest.java:18]: no need in unittests

File: frontend/src/components/jobGallery/useJob.tsx [hunk 6]
```diff
@@ -0,0 +1,24 @@
+import { useState } from 'react';
+import axios from 'axios';
+import { Jobs } from './model/Jobs';
+
+const useJobs = (): [Jobs[], () => void] => {
```
Comment (Kreyno93) [inline frontend/src/components/jobGallery/useJob.tsx:5]: thats confusing ~_~

**Review by Kreyno93 — APPROVED** [review]

File: backend/src/test/java/de/neuefische/backend/service/JobServiceTest.java (hunk 5 above)
Comment (KitNaIn) [reply backend/src/test/java/de/neuefische/backend/service/JobServiceTest.java:18]: removed annotation

File: backend/src/test/java/de/neuefische/backend/model/JobTest.java (hunk 3 above)
Comment (KitNaIn) [reply backend/src/test/java/de/neuefische/backend/model/JobTest.java:95]: removed model test

File: backend/src/test/java/de/neuefische/backend/service/GenerateUUIDServiceTest.java (hunk 4 above)
Comment (KitNaIn) [reply backend/src/test/java/de/neuefische/backend/service/GenerateUUIDServiceTest.java:10]: aMW

File: backend/src/test/java/de/neuefische/backend/model/JobTest.java (hunk 2 above)
Comment (KitNaIn) [reply backend/src/test/java/de/neuefische/backend/model/JobTest.java:79]: removed aswell

**Review by Kreyno93 — APPROVED** [review]

**Pull request merged**